### PR TITLE
Improve GetRemoteStatus and db.GetStatus() logic

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
- removes chance of db.GetStatus() recursion
- improves GetRemoteStatus() to require specifying if parent / children are required in status
- fix StatusCache to return object copies, and only store information about status itself (not related attachments)
- set `cache=shared` by default on SQLite instances to ensure thread-safety between operations (this was the source of the `nested transaction` related errors we were seeing). why this was not set as a default in a Go library where you have to go out of your way to run single-threaded?! who knows

### CODE REVIEW

**Comment by NyaaaWhatsUpDoc** [comment]
Snuck in another bugfix with a panic caused by EnrichRemoteAccount :)

**Review by tsmethurst — CHANGES_REQUESTED** [review]
Just the testing stuff needs to change, the rest looks great :+1:

File: internal/db/bundb/account.go [hunk 1]
```diff
@@ -45,60 +47,80 @@ func (a *accountDB) newAccountQ(account *gtsmodel.Account) *bun.SelectQuery {
...
+		},
+		func(account *gtsmodel.Account) error {
+			return a.newAccountQ(account).Where("account.url = ?", url).Scan(ctx)
+		},
+	)
 }
 
-func (a *accountDB) GetAccountByURL(ctx context.Context, uri string) (*gtsmodel.Account, db.Error) {
-	account := new(gtsmodel.Account)
+func (a *accountDB) getAccount(ctx context.Context, cacheGet func() (*gtsmodel.Account, bool), dbQuery func(*gtsmodel.Account) error) (*gtsmodel.Account, db.Error) {
```
[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/account.go:85]: Yep I will handle that when that PR is merged :)

File: internal/cache/account_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package cache_test
+
+import (
+	"testing"
+
+	"github.com/superseriousbusiness/gotosocial/internal/cache"
+	"github.com/superseriousbusiness/gotosocial/internal/gtsmodel"
+)
+
+func TestAccountCache(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/cache/account_test.go]: Done :+1:

File: internal/db/bundb/bundb.go [hunk 3]
```diff
@@ -91,14 +91,23 @@ func NewBunDBService(ctx context.Context, c *config.Config, log *logrus.Logger)
 		conn = WrapDBConn(bun.NewDB(sqldb, pgdialect.New()), log)
 	case dbTypeSqlite:
 		// SQLITE
+
+		// Drop anything fancy from DB address
```
[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/bundb.go:95]: I'll add an issue for it

File: internal/db/bundb/relationship.go [hunk 4]
```diff
@@ -298,7 +298,7 @@ func (r *relationshipDB) GetAccountFollowedBy(ctx context.Context, accountID str
 
 	if localOnly {
 		q = q.ColumnExpr("follow.*").
-			Join("JOIN accounts AS a ON follow.account_id = TEXT(a.id)").
+			Join("JOIN accounts AS a ON follow.account_id = CAST(a.id as TEXT)").
```
[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/relationship.go:301]: I just added tests for this (and the other relationship SQL change) to make sure!

[the remark this answers is not in the document]
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/relationship.go:301]: The tests seem to pass! As long as you can confirm these tests look good enough to check the SQL syntax I'd say this is ready :D

File: internal/db/bundb/relationship_test.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+func (suite *RelationshipTestSuite) TestIsMutualFollowing() {
+	suite.Suite.T().Skip("TODO: implement")
+}
+
+func (suite *RelationshipTestSuite) AcceptFollowRequest() {
+	for _, account := range suite.testAccounts {
+		_, err := suite.db.AcceptFollowRequest(context.Background(), account.ID, "NON-EXISTENT-ID")
+		if err != nil && !errors.Is(err, db.ErrNoEntries) {
```
Comment (tsmethurst) [inline internal/db/bundb/relationship_test.go:82]: you can just do `suite.Error(err)` here to assert an error, or even better do `suite.EqualError(err, db.ErrNoEntries)`

File: internal/db/bundb/relationship_test.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+func (suite *RelationshipTestSuite) GetAccountFollowedBy() {
+	// TODO: more comprehensive tests here
+
+	for _, account := range suite.testAccounts {
+		var err error
+
+		_, err = suite.db.GetAccountFollowedBy(context.Background(), account.ID, false)
+		if err != nil {
```
Comment (tsmethurst) [inline internal/db/bundb/relationship_test.go:107]: `suite.NoError(err)`

File: internal/db/bundb/relationship_test.go [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+			suite.Suite.Fail("error checking accounts followed by: %v", err)
+		}
+
+		_, err = suite.db.GetAccountFollowedBy(context.Background(), account.ID, true)
+		if err != nil {
```
Comment (tsmethurst) [inline internal/db/bundb/relationship_test.go:112]: `suite.NoError(err)`

File: internal/db/bundb/relationship_test.go [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+	testrig.StandardDBSetup(suite.db, suite.testAccounts)
+}
+
+func (suite *RelationshipTestSuite) TearDownTest() {
+	testrig.StandardDBTeardown(suite.db)
+}
+
+func (suite *RelationshipTestSuite) TestIsBlocked() {
+	suite.Suite.T().Skip("TODO: implement")
```
Comment (tsmethurst) [inline internal/db/bundb/relationship_test.go:60]: for all these I think you can just do `suite.T()` instead of `suite.Suite.T()`

**Review by tsmethurst — APPROVED** [review]

**Pull request merged**